model/task_lifecycle.go

```diff
@@ -1017,7 +1033,11 @@ func UpdateDisplayTask(t *task.Task) error {
 		}
 	}
 
```

Comment:
It might be useful to add a message here that logs the task ID, hasFailedTasks, unfinishedTasks, blockedTasks, and anything else that might be helpful to debug this in production if it looks like it's not working the way we expect.